source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -491,22 +494,52 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
     return false;
   }
   for (const GENERAL_NAME* san : san_names.get()) {
-    if (san->type == GEN_DNS) {
+    switch (san->type) {
+    case GEN_DNS: {
       ASN1_STRING* str = san->d.dNSName;
       const char* dns_name = reinterpret_cast<const char*>(ASN1_STRING_data(str));
       for (auto& config_san : subject_alt_names) {
         if (dNSNameMatch(config_san, dns_name)) {
           return true;
         }
       }
-    } else if (san->type == GEN_URI) {
+      break;
+    }
+    case GEN_URI: {
       ASN1_STRING* str = san->d.uniformResourceIdentifier;
       const char* uri = reinterpret_cast<const char*>(ASN1_STRING_data(str));
       for (auto& config_san : subject_alt_names) {
         if (config_san.compare(uri) == 0) {
           return true;
         }
       }
+      break;
+    }
+    case GEN_IPADD: {
+      sockaddr_storage ss;
```

Comment:
nit: You don't need `sockaddr_storage` here and reinterpret_cast it below. In the if blocks you can use `sockaddr_in{,6}` directly?